server/controllers/chartsController/helpers.js

```diff
@@ -303,3 +331,84 @@ export const calculateSubjectVariableDayCount = (
 
   return subjectsDayDataAndVariable.length
 }
+
+export const addSubjectsToResultsFromFiltersMap = ({
+  subjectsInFiltersA,
+  subjectsInFiltersB,
+  mapForResultsFromFilters,
+}) => {
+  subjectsInFiltersA.filter((subjectData) => {
+    const foundSubject = subjectsInFiltersB.find(
+      ({ subject }) => subject === subjectData.subject
+    )
+    const isSubjectInMap = mapForResultsFromFilters?.has(subjectData.subject)
+
+    if (foundSubject?.subject) {
+      if (isSubjectInMap) {
+        const subjectDataFromMap = mapForResultsFromFilters.get(
+          subjectData.subject
+        )
+
+        if (
+          subjectDataFromMap.socioDemographicsCollection &&
+          subjectDataFromMap.inclusionCriteriaCollection
+        )
+          return false
+        else {
+          mapForResultsFromFilters.set(subjectData.subject, {
+            ...subjectDataFromMap,
+            ...foundSubject,
+          })
+        }
+      } else {
+        mapForResultsFromFilters?.set(subjectData.subject, {
+          ...subjectData,
+          ...foundSubject,
+        })
+      }
+    } else return false
+  })
```

Comment:
I _think_ this whole function could be simplified pretty significantly. Would this result in the same outcome?

```js
const filterASubjects = new Set(subjectsInFiltersA.map(data => data.subject))
const filterBSubjects = new Set(subjectsInFiltersB.map(data => data.subject))

return new Set([...filterASubjects].filter((subject) => filterBSubjects.has(subject)))
// or return [...filterASubjects].filter((subject) => filterBSubjects.has(subject)) if we don't want the result to be a set
```

While this might not be _exactly_ the same, it should give us the ids of the subjects that are in both filters, which is I think what we want (the old `filteredSubjects` constant).

Reply:
https://github.com/AMP-SCZ/dpdash/pull/280/commits/c30a7d514c931ad36f83916c440ff7812449f315